libfastpm/solver.c

```diff
@@ -27,10 +27,29 @@ void fastpm_solver_init(FastPMSolver * fastpm,
 
     fastpm->config[0] = *config;
 
-    fastpm->cosmology[0] = (FastPMCosmology) {
-        .OmegaM = config->omega_m,
-        .OmegaLambda = 1.0 - config->omega_m,
-    };
+    if(!(config->cosmology)) {
+        /* Use a default fiducial cosmology. */
```

Comment:
Do you recall if we still hit this branch anywhere?

If not, we can probably convert this to a fastpm_raise(). 
This branch would cause a memory error anywy, because FDinterp is allocated on the stack, and we only copy an address of it to fastpm->cosmology. When the stack frame is destroyed, fastpm->cosmology->FDinterp points to an invalid address.